src/overlays/actors/ovl_En_Demo_heishi/z_en_demo_heishi.c

```diff
@@ -31,37 +39,150 @@ const ActorInit En_Demo_heishi_InitVars = {
+
+    this->headRotXTarget = 0;
+    if ((this->actor.xzDistToPlayer < 200.0f) && (yaw < 20000)) {
+        this->headRotXTarget = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+        if (this->headRotXTarget > 10000) {
+            this->headRotXTarget = 10000;
+        } else if (this->headRotXTarget < -10000) {
+            this->headRotXTarget = -10000;
+        }
+    }
+}
+
+void EnDemoheishi_SetupIdle(EnDemoheishi* this) {
+    EnDemoheishi_ChangeAnimation(this, 0);
+    this->actor.textId = sTextIds[this->unk272 = 0];
+    this->unk270 = 0;
+    this->actionFunc = EnDemoheishi_Idle;
+}
+
+void EnDemoheishi_Idle(EnDemoheishi* this, GlobalContext* globalCtx) {
+    s32 absYawDiff;
+    s16 yawDiff;
+
+    this->actor.flags &= 0xF7FFFFFF;
+    yawDiff = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    if (yawDiff < 0) {
+        absYawDiff = -yawDiff;
+    } else {
+        absYawDiff = yawDiff;
+    }
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        EnDemoheishi_SetupTalk(this);
+    } else if (absYawDiff <= 0x4BB8) {
+        func_800B8614(&this->actor, globalCtx, 70.0f);
+    }
+}
+
+void EnDemoheishi_SetupTalk(EnDemoheishi* this) {
+    this->unk270 = 1;
+    this->actionFunc = EnDemoheishi_Talk;
+}
+
+void EnDemoheishi_Talk(EnDemoheishi* this, GlobalContext* globalCtx) {
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && (func_80147624(globalCtx) != 0)) {
+        func_801477B4(globalCtx);
+        EnDemoheishi_SetupIdle(this);
+    }
+}
+
+void EnDemoheishi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderCylinder* collider;
+    EnDemoheishi* this = (EnDemoheishi*)thisx;
+
+    SkelAnime_Update(&this->skelAnime);
+    if (this->unk268 != 0) {
+        this->unk268--;
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1DU);
+    Actor_SetScale(&this->actor, 0.01f);
+    EnDemoheishi_SetHeadRotation(this);
+
+    Actor_SetHeight(&this->actor, 60.0f);
+    Math_SmoothStepToS(&this->headRotX, this->headRotXTarget, 1, 0xBB8, (s16)0);
+    Math_SmoothStepToS(&this->headRotY, this->headRotYTarget, 1, 0x3E8, (s16)0);
+    collider = &this->colliderCylinder;
+    Collider_UpdateCylinder(&this->actor, &this->colliderCylinder);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder.base);
+}
+
+s32 EnDemoheishi_OverrideLimbDraw(GlobalContext* globalctx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                  Actor* thisx) {
+    EnDemoheishi* this = THIS;
+
+    if (limbIndex == 16) {
+        rot->x += this->headRotX;
+        rot->y += this->headRotY;
+        rot->z += this->headRotZ;
+    }
+
+    return false;
+}
+
+void EnDemoheishi_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDemoheishi* this = (EnDemoheishi*)thisx;
```

Comment:
```suggestion
    EnDemoheishi* this = THIS;
```
Ah we haven't decided yet when/if we will remove the THIS macro, so its still to be used for now.

Reply:
Fixed, and good to know. I was actually wondering about this.